src/main/kotlin/solve/importer/ProjectParser.kt

```diff
@@ -110,7 +114,61 @@ object ProjectParser {
             )
         }
         landmarks.forEach { (name, outputs) ->
-            frames.add(ProjectFrame(name, Path(images[name]!!), outputs))
+            try {
+                val longName = name.toLong()
+                frames.add(ProjectFrame(longName, Path(images[name]!!), outputs))
+            } catch (e: Exception) {
+                frames.add(
+                    ProjectFrame(
+                        0,
+                        Path(images[name]!!),
+                        outputs,
+                        errorMsg = mutableListOf("$name can not converted to Long"),
+                        errorType = mutableListOf(ErrorType.INCORRECT_IMAGE_NAME)
+                    )
+                )
+            }
+        }
+        frames.forEach {
+            it.landmarkFiles.forEach { a ->
+                when (a.projectLayer.kind) {
+                    LayerKind.Plane ->
+                        try {
+                            ImagePlanesParser.parse(a.path.toString())
+                        } catch (e: Exception) {
+                            it.errorType.add(ErrorType.ERROR_PARSER)
+                            it.errorMsg.add("Couldn't parse ${a.projectLayer}")
+                        }
+
+                    LayerKind.Keypoint ->
+                        try {
+                            CSVPointsParser.parse(a.path.toString())
+                        } catch (e: Exception) {
+                            it.errorType.add(ErrorType.ERROR_PARSER)
+                            it.errorMsg.add("Couldn't parse ${a.projectLayer}")
+                        }
+
+                    LayerKind.Line ->
+                        try {
+                            CSVLinesParser.parse(a.path.toString())
+                        } catch (e: Exception) {
+                            it.errorType.add(ErrorType.ERROR_PARSER)
+                            it.errorMsg.add("Couldn't parse ${a.projectLayer}")
+                        }
+                }
+            }
+            if (!(it.imagePath.extension != ".png" && it.imagePath.extension != ".jpg")) {
```

Comment:
Add set of possible extensions and check if is contains `it.imagePath.extension `